### PR TITLE
Implement late time projection in TimesNet

### PR DESCRIPTION
**Pull request by ShinDongWoon**
## Summary
- replace the pre-extension linear layer with a dedicated forecast_time_proj head that is initialized to copy the latest observation
- run TimesBlocks over the historical window only and feed their output through the new late time-projection path for mean and scale
- add safety assertions around the projection head and output shapes while keeping the late bias head plumbing intact

## Testing
- python - <<'PY'
import torch, sys
sys.path.append('src')
from timesnet_forecast.models.timesnet import TimesNet

B, L, H, N = 2, 16, 4, 3
model = TimesNet(
    input_len=L,
    pred_len=H,
    d_model=8,
    d_ff=16,
    n_layers=1,
    k_periods=1,
    kernel_set=[(3, 3)],
    dropout=0.1,
    activation='gelu',
    mode='direct',
)

x = torch.randn(B, L + 5, N)
mu, sigma = model(x)
print('mu shape', mu.shape)
print('sigma shape', sigma.shape)
print('sigma min', sigma.min().item())
PY

------
https://chatgpt.com/codex/tasks/task_e_68d4c62b02608328aa6810d57b117617

### CODE REVIEW

**Comment by ShinDongWoon** [comment]
@codex review
